src/iminuit/minuit.py

```diff
@@ -2347,10 +2352,29 @@ def interactive(
         --------
         Minuit.visualize
         """
-        from iminuit.ipywidget import make_widget
+        try:
+            from IPython import get_ipython
+
+            if (
+                get_ipython().__class__.__name__ == "ZMQInteractiveShell"
+                and "IPKernelApp" in get_ipython().config
+            ):
+                is_jupyter = True
+            else:
+                is_jupyter = False
+        except Exception:
+            is_jupyter = False
```

Comment:
This needs to be factored out into an utility function called "is_jupyter", please put in utils.

Reply:
When you do that, can you please also put a reference for this detection magic? I wonder how brittle this is. The ZMQ prefix looks suspiciously like an implementation detail.

Reply:
I did some research. ChatGPT gives your method as the first of several options, as does Stackoverflow. Digging a bit deeper into the SO answers, I found this technique, which I like slightly better:
https://github.com/tqdm/tqdm/blob/master/tqdm/autonotebook.py

```
def is_jupyter_notebook():
    try:
        from IPython import get_ipython
        if 'IPKernelApp' not in get_ipython().config:  # pragma: no cover
            return False
    except ImportError:
        return False
    except AttributeError:
        return False
    return True
```

This also seems to depend on some implementation details, but those seem less likely to change.

Reply:
Here is another option from IPython issues, although not confirmed by the IPython devs https://github.com/ipython/ipython/issues/9732

```py
def is_jupyter_notebook():
    try:
         from IPython import get_ipython
         ip = get_ipython()
         return ip.has_trait('kernel')
    except ImportError:
         return False
    return False
```

Reply:
I went with the second one for now and moved it into util.